src/schema/comments.ts

```diff
@@ -284,6 +313,86 @@ export interface GQLUserCommentsArgs extends ConnectionArguments {
   userId: string;
 }
 
+interface RecentMentionsProps {
+  name?: string;
+  limit?: number;
+}
+
+interface MentionedUser {
+  id: string;
+  username?: string;
+}
+
+const getVerifiedMentions = async (
+  con: ORMConnection | EntityManager,
+  usernames: string[],
+  userId: string,
+): Promise<MentionedUser[]> =>
+  con
+    .getRepository(User)
+    .find({ where: { username: In(usernames), id: Not(userId) } });
+
+const getMentions = async (
+  con: ORMConnection | EntityManager,
+  content: string,
+  userId: string,
+): Promise<MentionedUser[]> => {
+  const words = content.split(' ');
+  const result = words
+    .filter(([first, next]) => first === '@' && !!next)
+    .map((mention) => mention.substring(1));
+
+  if (result.length === 0) {
+    return [];
+  }
+
+  return getVerifiedMentions(con, result, userId);
+};
+
+const saveCommentMentions = (
+  transaction: EntityManager,
+  commentId: string,
+  users: MentionedUser[],
+) => {
+  if (!users.length) return;
+
+  const mentions: Partial<CommentMention>[] = users.map(({ id }) => ({
+    commentId,
+    mentionedUserId: id,
+  }));
+
+  return transaction
+    .createQueryBuilder()
+    .insert()
+    .into(CommentMention)
+    .values(mentions)
+    .orIgnore()
+    .execute();
+};
+
+export const getRecentMentions = (
+  con: ORMConnection,
+  userId: string,
+  { limit = 5, name }: RecentMentionsProps,
+): Promise<GQLMentionUser[]> => {
+  let query = con
+    .getRepository(CommentMention)
+    .createQueryBuilder('cm')
+    .select('u.name, u.username, u.image')
+    .innerJoin(Comment, 'c', 'cm."commentId" = c.id')
+    .innerJoin(User, 'u', 'u.id = cm."mentionedUserId"')
+    .where('c."userId" = :userId', { userId })
+    .limit(limit);
```

Comment:
What users should this query return?

Reply:
It is the users that were recently mentioned by the requesting user.